### PR TITLE
fix: install terraform in base Docker image

### LINKED ISSUE #6258: bug: in air-gapped kubernetes cluster, coderd deployment crashloops
Observed during customer call: get UI error "something went wrong" intermittently.
Appears to only happen on air-gapped cluster that cannot access internet.

Steps to reproduce (reproduced on OpenShift 4.12 single-node):
- Install Coder in Kubernetes via Helm Chart
- Create NetworkPolicies that only allow traffic within the cluster
- Observe coder pod crashing periodically with following message:
  ```
  Unexpected error, shutting down server: install terraform: install: Get "https://releases.hashicorp.com/terraform/1.3.4/index.json": dial tcp 143.204.68.20:443: i/o timeout
  ```

### PR DESCRIPTION
**Pull request by johnstcn**
This PR adds terraform to all Coder base images. It's a required dependency so it makes sense to bundle it with the image. 

~Still TODO: providers and plugins~
Not bundling providers at this time.

Fixes https://github.com/coder/coder/issues/6258

### CODE REVIEW

**Review by kylecarbs — APPROVED** [review]

**Comment by coadler** [comment]
I think there's a weird interaction with the Terraform downloader such that it will still try to download Terraform when it's installed but the wrong version. Are we able to be sure those versions stay in sync?

**Comment by johnstcn** [comment]
> I think there's a weird interaction with the Terraform downloader such that it will still try to download Terraform when it's installed but the wrong version. Are we able to be sure those versions stay in sync?

We could specify with our `apk add` version == the Terraform version we try to fetch? How picky is it?

**Comment by bpmct** [comment]
https://github.com/coder/coder/blob/19ae411f05b6a23d9336938cba1bd632c6116b2b/provisioner/terraform/install.go#L21-L24

**Comment by johnstcn** [comment]
> https://github.com/coder/coder/blob/19ae411f05b6a23d9336938cba1bd632c6116b2b/provisioner/terraform/install.go#L21-L24

Seems like the simplest option is to manually specify the version in `apk` but add comments both in the base image and in the above file to ensure these are kept in sync.

File: scripts/Dockerfile.base [hunk 1]
```diff
@@ -22,6 +22,25 @@ RUN apk add --no-cache \
 		-G coder \
 		coder
 
+# Install Terraform plugins
+RUN mkdir -p /opt/terraform/plugins/registry.terraform.io
+# Add config for local terraform
+ADD files/terraform-config.tfrc /opt/terraform/config.tfrc
+ARG CODER_PROVIDER_VERSION=0.6.12
```
Comment (bpmct) [inline scripts/Dockerfile.base:29]: These versions (for Coder and the provider) can also get quickly out of sync from what we hardcode in our example templates too, which will cause Terraform to still fail

Maybe we stick to just embedding Terraform and make the TRFC configurable via a coder-server flag or something? Then, an external PVC can be mounted or a network mirror can be specified https://github.com/coder/coder/issues/6189

File: scripts/files/terraform-config.tfrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+provider_installation {
+  filesystem_mirror {
+    path = "/opt/terraform/plugins"
+  }
+}
```
Comment (bpmct) [inline scripts/files/terraform-config.tfrc:5]: Will Coder attempt to download from the public registry if there isn't something in this mirror, like our other cloud examples? I genuinely don't know.

File: scripts/Dockerfile.base (hunk 1 above)
Comment (johnstcn) [reply scripts/Dockerfile.base:29]: Yeah, that sounds like a better option. Baking in all of the Terraform providers we use in `examples/templates` would inflate our base image by 199MB.

**Review by bpmct — COMMENTED** [review]
If you are going to merge this, can you change our [offline docs](https://coder.com/docs/v2/latest/install/offline#offline-container-images)?

**Comment by johnstcn** [comment]
> If you are going to merge this, can you change our [offline docs](https://coder.com/docs/v2/latest/install/offline#offline-container-images)?

Yep

File: scripts/files/terraform-config.tfrc (hunk 2 above)
Comment (johnstcn) [reply scripts/files/terraform-config.tfrc:5]: I'm not sure either; so I'm going to omit this change.

**Review by bpmct — APPROVED** [review]

**Pull request merged**